velox/common/memory/MemoryPool.cpp

```diff
@@ -974,6 +980,9 @@ void MemoryPoolImpl::testingSetCapacity(int64_t bytes) {
 }
 
 bool MemoryPoolImpl::needRecordDbg(bool isAlloc) {
+  if (!debugRegexFilter_.empty()) {
+    return RE2::FullMatch(name_, debugRegexFilter_);
+  }
   // TODO(jtan6): Add sample based condition support.
```

Comment:
Does TODO still apply? Can we remove isAlloc if not required for now?

Reply:
Yeah it applies. I'll add some sample based functionality in a followup

Reply:
isAlloc is also required?

Reply:
Yeah it will be required for sample based.